apps/api/src/modules/report/infrastructure/queries/retrieve-report.query.ts

```diff
@@ -0,0 +1,201 @@
+    const dossierData = this.parseDossierContent(dossier.content);
+    const rules = this.buildRulesObject(report.reportRules);
+    const attachedFiles = this.buildAttachedFiles(report.files);
+
+    return {
+      id: report.id,
+      sessionId: report.sessionId,
+      state: report.state,
+      formation: report.formation,
+      comment: report.comment,
+      transparency: session.name,
+      dateTransparence: sessionContent.dateTransparence,
+      attachedFiles,
+      rules,
+      ...dossierData,
+    };
+  }
+
+  private parseSessionContent(content: any) {
+    return z.object({ dateTransparence: dateOnlyJsonSchema }).parse(content);
+  }
+
+  private parseDossierContent(content: any) {
+    const normalized = z.parse(NominationFileContentSchema, content);
+
+    return {
+      name: normalized.nomMagistrat,
+      birthDate: normalized.dateDeNaissance,
+      grade: normalized.grade,
+      biography: normalized.historique,
+      currentPosition: normalized.posteActuel,
+      targettedPosition: normalized.posteCible,
+      rank: normalized.rang,
+      folderNumber: normalized.numeroDeDossier,
+      dueDate: normalized.dateEchéance,
+      observers: normalized.observants,
+      dureeDuPoste: normalized.datePriseDeFonctionPosteActuel
+        ? this.calculateDureeDuPoste(normalized.datePriseDeFonctionPosteActuel)
+        : null,
+    };
+  }
+
+  private calculateDureeDuPoste(datePriseDeFonction: any): string {
+    // Convertir en string si c'est un objet DateOnly
+    const dateString =
+      typeof datePriseDeFonction === 'string'
+        ? datePriseDeFonction
+        : `${datePriseDeFonction.year}-${String(datePriseDeFonction.month).padStart(2, '0')}-${String(datePriseDeFonction.day).padStart(2, '0')}`;
+
+    const dateObj = new Date(dateString);
+    const now = new Date();
+    const months =
+      (now.getFullYear() - dateObj.getFullYear()) * 12 +
+      (now.getMonth() - dateObj.getMonth());
+
+    const years = Math.floor(months / 12);
+    const remainingMonths = months % 12;
+
+    if (years === 0) return `${remainingMonths} mois`;
+    if (remainingMonths === 0) return `${years} an${years > 1 ? 's' : ''}`;
+    return `${years} an${years > 1 ? 's' : ''} et ${remainingMonths} mois`;
+  }
+
+  private buildRulesObject(
+    reportRules: Array<{
+      id: string;
+      ruleGroup: string;
+      ruleName: string;
+      validated: boolean;
+    }>,
+  ): Record<string, Record<string, any>> {
+    const rules: Record<string, Record<string, any>> = {};
+
+    for (const rule of reportRules) {
+      if (!rules[rule.ruleGroup]) {
+        rules[rule.ruleGroup] = {};
+      }
+
+      rules[rule.ruleGroup]![rule.ruleName] = {
+        id: rule.id,
+        validated: rule.validated,
+        preValidated: false,
+      };
+    }
+
+    return rules;
+  }
+
+  private buildAttachedFiles(
+    files: Array<{ usage: string; file: { id: string; name: string } }>,
```

Comment:
nit-pick:

```suggestion
    files: { usage: string; file: { id: string; name: string } }[],
```

Et même ici

```suggestion
    files: readonly { usage: string; file: { id: string; name: string } }[],
```